src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Weather/CSSISpaceWeather.cpp

```diff
@@ -0,0 +1,588 @@
+    }
+}
+
+const Interval& CSSISpaceWeather::accessMonthlyPredictionInterval() const
+{
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("CSSI Space Weather");
+    }
+
+    return monthlyPredictionInterval_;
+}
+
+const CSSISpaceWeather::MonthlyPrediction& CSSISpaceWeather::accessMonthlyPredictionAt(const Instant& anInstant) const
+{
+    if (!anInstant.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Instant");
+    }
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("CSSI Space Weather");
+    }
+
+    if (!monthlyPredictionInterval_.contains(anInstant))
+    {
+        throw ostk::core::error::RuntimeError(
+            "Instant [{}] out of prediction range [{} - {}].",
+            anInstant.toString(Scale::UTC),
+            monthlyPredictionInterval_.accessStart().toString(Scale::UTC),
+            monthlyPredictionInterval_.accessEnd().toString(Scale::UTC)
+        );
+    }
+
+    const Integer year = anInstant.getDateTime(Scale::UTC).getDate().getYear();
+    const Integer month = anInstant.getDateTime(Scale::UTC).getDate().getMonth();
+
+    const Real monthMjd = DateTime(year, month, 1).getModifiedJulianDate();
+
+    const auto predictionIt = monthlyPredictions_.find(monthMjd.floor());
+
+    if (predictionIt != monthlyPredictions_.end())
+    {
+        return predictionIt->second;
+    }
+    else
+    {
+        throw ostk::core::error::RuntimeError("Cannot find prediction at [{}].", anInstant.toString(Scale::UTC));
+    }
+}
+
+CSSISpaceWeather CSSISpaceWeather::Undefined()
+{
+    return CSSISpaceWeather();
+}
+
+CSSISpaceWeather CSSISpaceWeather::Load(const fs::File& aFile)
+{
+    using ostk::core::types::Index;
+    using ostk::core::types::Uint8;
+    using ostk::core::types::Uint16;
+    using ostk::core::types::Real;
+    using ostk::core::types::String;
+
+    using ostk::physics::time::Scale;
+    using ostk::physics::time::Time;
+    using ostk::physics::time::DateTime;
+
+    if (!aFile.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("File");
+    }
+
+    if (!aFile.exists())
+    {
+        throw ostk::core::error::RuntimeError("File [{}] does not exist.", aFile.toString());
+    }
+
+    CSSISpaceWeather spaceWeather;
+    std::ifstream fileStream {aFile.getPath().toString()};
+
+    String line;
+
+    auto split = [](String s, const String delimiter) -> Array<String>
+    {
+        Array<String> results;
+
+        if (s.empty())
+        {
+            return results;
+        }
+
+        size_t last = 0;
+        size_t next = 0;
+        while ((next = s.find(delimiter, last)) != std::string::npos)
+        {
+            results.add(s.substr(last, next - last));
+            last = next + 1;
+        }
+
+        results.add(s.substr(last));
+        return results;
+    };
+
+    while (std::getline(fileStream, line))
```

Comment:
Could you use `Table::Load(aFile, Table::Format::CSV, true)` and then maybe loop through the rows?

Reply:
from ostk::core::ctnr::Table?

Reply:
Yeah I realized that the `Table` class existed only after writing this 😅. Totally can do that. I'm not sure how much it buys us? I don't know if it would reduce the code needed here by much. Thoughts?

Reply:
I think it helps abstract the File I/O and also it removes the need for `lexical_cast` as it converts objects to String, Real and Int appropriately. I would say it's worth it as it might reduce a fair bit of code but I could be wrong. I'm happy to open a suggestion MR.

Reply:
Hmm if it removes the lexical cast, I am all for it 🙏. Good suggestion, I'll try it.